content/docs/extension-author-guide/create-extension/build-dockerfile.md

```diff
@@ -0,0 +1,80 @@
++++
+title="Generating a build.Dockerfile"
+weight=404
++++
+
+### Examine `tree` extension
+
+#### detect
+
+`cat $workspace/samples/extensions/tree/bin/detect` - the extension always detects (because its exit code is `0`) and provides a dependency
+  called `tree` by writing to the build plan
+
+#### generate
+
+`cat $workspace/samples/extensions/tree/bin/generate` - the extension generates a `build.Dockerfile` that installs `tree` on the builder
+  image
+
+### Re-create our builder with `hello-extensions` updated to require `tree`
+
+1. Edit `$workspace/samples/buildpacks/hello-extensions/bin/detect` to uncomment the first set of lines that
+  output `[[requires]]` to the build plan
+
+2. Create the builder:
+
+```
+$workspace/pack/out/pack builder create $registry_namespace/extensions-builder \
+  --config $workspace/samples/builders/alpine/builder.toml \
+  --publish
+```
+
+### Build the application image
+
+```
+$workspace/pack/out/pack build hello-extensions \
+  --builder $registry_namespace/extensions-builder \
+  --lifecycle-image $LIFECYCLE_IMAGE \
+  --verbose
+```
+
+You should see:
+
+```
+[detector] ======== Results ========
```

Comment:
Could we add Ugo tests for these things if possible? This ensures that the docs don't drift from what is happening in pack.

Reply:
Sure, but could I get a little guidance in setting that up? I'm not super well versed in Ugo.

Reply:
@samj1912 ping

Reply:
https://github.com/jromero/ugo is a mechanism to add tests to markdown docs.  A good example is https://raw.githubusercontent.com/buildpacks/docs/main/content/docs/buildpack-author-guide/create-buildpack/build-app.md where we create a `suite`, and then add some `file` contents, `exec` some commands and then test the output of the command executions.

Reply:
Wow, I really didn't know when I'd see this day but this is added as of f5d2b3c4